src/components/Arping/Arping.tsx

```diff
@@ -0,0 +1,217 @@
+     * @param {string} data - The data received from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data);
+    }, []);
+
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            // If the process was successful, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+                // If the process was terminated manually, display a termination message.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+            } else {
+                // If the process was terminated with an error, display the exit and signal codes.
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+            // Allow Saving as the output is finalised
+            setAllowSave(true);
+            setHasSaved(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    /**
+     * Function to handle form submission.
+     * Executes the Arping component with the provided form values.
+     * @param {FormValuesType} values - Form values containing Arping input.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        // Activate loading state to indicate ongoing process.
+        setLoading(true);
+        // Disallow saving until the tool's execution is complete
+        setAllowSave(false);
+        // Construct arguments for the Arping component command based on form input.
+        let args = [values.targetIP];
+
+        if (values.count) {
+            args.push("-c", values.count);
+        }
+
+        if (values.interface) {
+            args.push("-I", values.interface);
+        }
+
+        if (values.verbose) {
+            args.push("-v");
+        }
+        //Execute the Arping component via helper method and handle its output or potential errors.
+        CommandHelper.runCommandWithPkexec("arping", args, handleProcessData, handleProcessTermination)
+            .then(() => {
```

Comment:
Need to insert these variables to ensure the command is displayed on the output block. 
```suggestion
            .then(({ output, pid }) => {
```